src/services/navigationBar.ts

```diff
@@ -2,856 +2,551 @@
-                        return member.kind === SyntaxKind.Constructor && member;
-                    });
-
-                    // Add the constructor parameters in as children of the class (for property parameters).
-                    // Note that *all non-binding pattern named* parameters will be added to the nodes array, but parameters that
-                    // are not properties will be filtered out later by createChildItem.
-                    const nodes: Node[] = removeDynamicallyNamedProperties(node);
-                    if (constructor) {
-                        addRange(nodes, filter(constructor.parameters, p => !isBindingPattern(p.name)));
-                    }
-
-                    childItems = getItemsWorker(sortNodes(nodes), createChildItem);
+                if (a.body.kind !== SyntaxKind.ModuleDeclaration) {
+                    return true;
                 }
-
-                const nodeName = !node.name ? "default" : node.name.text;
-
-                return getNavigationBarItem(
-                    nodeName,
-                    ts.ScriptElementKind.classElement,
-                    getNodeModifiers(node),
-                    [getNodeSpan(node)],
-                    childItems,
-                    getIndent(node));
+                return areSameModule(<ModuleDeclaration>a.body, <ModuleDeclaration>b.body);
             }
+        }
 
-            function createEnumItem(node: EnumDeclaration): ts.NavigationBarItem {
-                const childItems = getItemsWorker(sortNodes(removeComputedProperties(node)), createChildItem);
-                return getNavigationBarItem(
-                    node.name.text,
-                    ts.ScriptElementKind.enumElement,
-                    getNodeModifiers(node),
-                    [getNodeSpan(node)],
-                    childItems,
-                    getIndent(node));
+        /** Merge source into target. Source should be thrown away after this is called. */
+        function merge(target: NavNode, source: NavNode): void {
+            target.additionalNodes = target.additionalNodes || [];
+            target.additionalNodes.push(source.node);
+            if (source.additionalNodes) {
+                target.additionalNodes.push(...source.additionalNodes);
             }
 
-            function createInterfaceItem(node: InterfaceDeclaration): ts.NavigationBarItem {
-                const childItems = getItemsWorker(sortNodes(removeDynamicallyNamedProperties(node)), createChildItem);
-                return getNavigationBarItem(
-                    node.name.text,
-                    ts.ScriptElementKind.interfaceElement,
-                    getNodeModifiers(node),
-                    [getNodeSpan(node)],
-                    childItems,
-                    getIndent(node));
-            }
+            target.children.push(...source.children);
+            mergeChildren(target.children);
+            sortChildren(target.children);

```

Comment:
do you need to sort every time you merge? we are sorting once at the end anyways.. correct?